### PR TITLE
validation and tests

### PR DESCRIPTION
**Pull request by scottnath**
Initial plugin created for currency.

---

Resolves #https://github.com/punchcard-cms/punchcard/issues/179

`DCO 1.1 Signed-off-by: Scott Nath <github@scottnath.com>`
`DCO 1.1 Signed-off-by: Sam Richard <sam@snug.ug>`

### CODE REVIEW

**Comment by scottnath** [comment]
This is blocked waiting on changes that are in [this pr for eslint-config-punchcard](https://github.com/punchcard-cms/eslint-config-punchcard/issues/17). Until those changes are in npm, this input plugin won't pass CI.

**Comment by scottnath** [comment]
@Snugug I need another pair of eyes on this. Right now the dependencies in this module are _exactly the same_ as [input-plugin-boolean](https://github.com/punchcard-cms/input-plugin-boolean) and yet I'm still getting this error:

```
Cannot find module 'eslint-config-defaults/configurations/airbnb/es6'
Referenced from: punchcard/configurations/node
Referenced from: punchcard
Referenced from: /home/travis/build/punchcard-cms/input-plugin-currency/.eslintrc.yml
Error: Cannot find module 'eslint-config-defaults/configurations/airbnb/es6'
Referenced from: punchcard/configurations/node
Referenced from: punchcard
Referenced from: /home/travis/build/punchcard-cms/input-plugin-currency/.eslintrc.yml
```

It looks like this is an issue with `npm install` in travis.

Can you please look this over? I've been at this one for a long time and I'm clearly missing a setting or something. Story now weeks late.

File: index.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      validation: {
+        function: 'currencyValidation',
+        on: 'blur',
+      },
+      label: 'Please enter a currency',
+      placeholder: '42.42',
+      type: 'number',
+      settings: {
+        min: 0,
+        step: '0.01',
```
Comment (Snugug) [inline index.js:31]: Would like to see a setting (and output of) the currency symbol

**Comment by Snugug** [comment]
@scottnath Tests are passing now, got one comment, then GTG

**Comment by scottnath** [comment]
@Snugug symbol added

**Comment by scottnath** [comment]
@Snugug this is still not working for some reason btw. Travis is showing a [failing build](https://travis-ci.org/punchcard-cms/input-plugin-currency/builds/158805065) for the icon and it has not published to npm.

**Pull request merged**